pkg/cmd/cli/cmd/newapp.go

```diff
@@ -170,6 +176,16 @@ func RunNewApplication(f *clientcmd.Factory, out io.Writer, c *cobra.Command, ar
 				hasMissingRepo = true
 				fmt.Fprintf(c.Out(), "WARNING: We created an image stream %q, but it does not look like a Docker registry has been integrated with the OpenShift server. Automatic builds and deployments depend on that integration to detect new images and will not function properly.\n", t.Name)
 			}
+		case *deployapi.DeploymentConfig:
```

Comment:
The major reason was to print the generated parameter values (`MYSQL_USER`, `MYSQL_PASSWORD`, etc).